### PR TITLE
look ahead prepration

### PR DESCRIPTION
**Pull request by kianzarrin**
use ref instead of copying vehicle/citizen (faster and thread safe)
cleaned build script(so that I can build on my PC) and added solution (helps VS to manage sub modules)
expand transform methods. (I will use in future)
added render overlay util. (I will use in future)

### CODE REVIEW

File: CSkyL.csproj [hunk 1]
```diff
@@ -11,6 +11,11 @@
     <AssemblyName>CSkyL</AssemblyName>
     <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
+    <AssemblySearchPaths>
+      $(AssemblySearchPaths);
+      C:\Program Files (x86)\Steam\steamapps\common\Cities_Skylines\Cities_Data\Managed\;
+      ..\..\..\..\..\SteamLibrary\steamapps\common\Cities_Skylines\Cities_Data\Managed\
+    </AssemblySearchPaths>
```
Comment (Asu4ni) [inline CSkyL.csproj:18]: The modification of this project file somehow messed up my building process.

I've tried `git checkout main -- CSkyL.csproj` and simply put this `AssemblySearchPaths` part in and rebuild. These are the warnings/errors:

image

Do you know what happened here? cause I have no clues why simply inserting these lines caused those errors/warnings to be generated.

File: UI/OverlayUtil.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+﻿using ColossalFramework.Math;
```
Comment (Asu4ni) [inline UI/OverlayUtil.cs:1]: I've seen somebody suggesting putting those `using`s inside the namespace. But I'm not sure why though lol.

File: Transform.cs [hunk 3]
```diff
@@ -138,6 +159,16 @@ public Angle(float yawDegree, float pitchDegree)
 
         private static float _ToYaw(float angle) => angle.Modulus(rangeDegree);
         private static float _ToPitch(float angle) => angle.Clamp(rangePitch);
+
+        public Displacement ToDisplacement(float lenght) =>
+            Displacement._FromVec3(_AsQuat * Vector.forward * lenght);
+
+        public static Angle Lerp(Angle a, Angle b, float t)
```
[the remark this answers is not in the document]
Comment (kianzarrin) [reply Transform.cs:166]: I can understand there is difference between `Quaternion.Lerp` and `Quaternion.SLerp`.
But is there difference between `Quaternion.SLerp` and `Mathf.LerpAngle` ?
ill do as you say

File: UI/OverlayUtil.cs (hunk 2 above)
Comment (kianzarrin) [reply UI/OverlayUtil.cs:1]: I think there is a reason but I forgot. this was automatic code generation which is why its like this

File: Transform.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Asu4ni) [reply Transform.cs:166]: After some thought, let's consider this case.
```
a.yawDegree = 0, a.pitchDegree = 0
b.yawDegree = 90, a.pitchDegree = 89
t = .5
```
`Quaternion.Slerp` should do the job, which should outputs something like: `yaw = .5, pitch = 45`.
On the other hand, using `Mathf.LerpAngle` would result in: `yaw = 45, pitch = 44.5`

[the remark this answers is not in the document]
Comment (kianzarrin) [reply Transform.cs:166]: i thought yaw is horizontal angle and pitch is vertical. Obviously I was wrong.

File: CSkyL.csproj (hunk 1 above)
Comment (kianzarrin) [reply CSkyL.csproj:19]: ok fixed it.

**Pull request merged**